### PR TITLE
Vanilla Pike on Clear

### PR DESCRIPTION
**Pull request by ychen2u**
Signed-off-by: Yan Chen <yan.chen@intel.com>

### CODE REVIEW

**Comment by ychen2u** [comment]
Please dont merge now, still need to clean the code.

File: stackrc [hunk 1]
```diff
@@ -259,7 +259,8 @@ WSGI_MODE=${WSGI_MODE:-"uwsgi"}
 
 # Base GIT Repo URL
 # Another option is https://git.openstack.org
-GIT_BASE=${GIT_BASE:-git://git.openstack.org}
+#GIT_BASE=${GIT_BASE:-git://git.openstack.org}
+GIT_BASE=${GIT_BASE:-https://git.openstack.org}
```
Comment (dtroyer) [inline stackrc]: This is an issue due to Intel's proxy, handle it in configuration by setting GIT_BASE in local.conf

File: inc/python [hunk 2]
```diff
@@ -52,7 +52,8 @@ function get_python_exec_prefix {
     if is_fedora || is_suse; then
         echo "/usr/bin"
     else
-        echo "/usr/local/bin"
+        #echo "/usr/local/bin"
+        echo "/usr/bin"
```
Comment (dtroyer) [inline inc/python]: I am really curious why this?  This value is used for install locations and pointing at the system install is dangerous, even for something as carefree as DevStack.

File: functions-common [hunk 3]
```diff
@@ -1542,7 +1542,7 @@ function _run_under_systemd {
     fi
 
     $SYSTEMCTL enable $systemd_service
-    $SYSTEMCTL start $systemd_service
+    $SYSTEMCTL restart $systemd_service
```
Comment (dtroyer) [inline functions-common]: Heh, this is funny how many times this has been switched upstream.  I prefer using restart but some hate seeing the log messages when the service is not already running...

File: stackrc [hunk 4]
```diff
@@ -138,13 +138,15 @@ if [[ -r $RC_DIR/.localrc.password ]]; then
 fi
 
 # Control whether Python 3 should be used at all.
-export USE_PYTHON3=$(trueorfalse False USE_PYTHON3)
+#export USE_PYTHON3=$(trueorfalse False USE_PYTHON3)
+export USE_PYTHON3=False
```
Comment (dtroyer) [inline stackrc]: This should be set in local.conf, not changed here

File: stackrc [hunk 5]
```diff
@@ -138,13 +138,15 @@ if [[ -r $RC_DIR/.localrc.password ]]; then
...
 
 # Control whether Python 3 is enabled for specific services by the
 # base name of the directory from which they are installed. See
 # enable_python3_package to edit this variable and use_python3_for to
 # test membership.
-export ENABLED_PYTHON3_PACKAGES="nova,glance,cinder,uwsgi,python-openstackclient"
+#export ENABLED_PYTHON3_PACKAGES="nova,glance,cinder,uwsgi,python-openstackclient"
+export ENABLED_PYTHON3_PACKAGES=""
```
Comment (dtroyer) [inline stackrc]: This should be set in local.conf

File: functions-common [hunk 6]
```diff
@@ -312,6 +312,9 @@ declare -g os_VENDOR os_RELEASE os_PACKAGE os_CODENAME
 # user if not available.  Note can't use generic install_package*
 # because they depend on this!
 function _ensure_lsb_release {
+    # ignore lsb_release check
+    return
```
Comment (dtroyer) [inline functions-common]: Does Clear have anything similar to this at all?  We use it to get the values returned in GetOSVersion().  We can probably detect Clear by looking for swupd and simulating an equivalent for what we need.

Comment (ychen2u) [reply functions-common]: We can use "grep "Clear Linux" /lib/os-release".
I will try to make it work today.

File: stackrc (hunk 5 above)
Comment (ychen2u) [reply stackrc]: I will move it.

File: stackrc (hunk 4 above)
Comment (ychen2u) [reply stackrc]: I will move it.

File: inc/python (hunk 2 above)
Comment (ychen2u) [reply inc/python]: haha, same feeling, but this is the reality.
There are also several other location issues like this, in neutron/glance configs.

File: stackrc (hunk 1 above)
Comment (ychen2u) [reply stackrc]: Ok, I will move it out.

File: inc/python (hunk 2 above)
Comment (dtroyer) [reply inc/python]: I am curious as to why... is /usr/local forbidden?  These are things that are _not_ managed by the OS packaging system, why must they mingle with those?  That is basic sysadmin 101 stuff...

File: functions [hunk 7]
```diff
@@ -745,8 +745,13 @@ function set_systemd_override {
...
     local value="$2"
 
-    local sysconf="/etc/systemd/system.conf"
-    iniset -sudo "${sysconf}" "Manager" "$key" "$value"
+    #local sysconf="/etc/systemd/system.conf"
+    #iniset -sudo "${sysconf}" "Manager" "$key" "$value"
+    local sysconf_40="/usr/lib/systemd/system.conf.d/40-core-ulimit.conf"
+    iniset -sudo "${sysconf_40}" "Manager" "$key" "$value"
+    local sysconf_50="/usr/lib/systemd/system.conf.d/50-nfiles.conf"
+    iniset -sudo "${sysconf_50}" "Manager" "$key" "$value"
```
Comment (dtroyer) [inline functions]: These files do not exist on the clear install I am looking at, should these go into /etc?

File: tools/fixup_stuff.sh [hunk 8]
```diff
@@ -202,5 +202,5 @@ fi
 # on python-virtualenv), first install the distro python-virtualenv
 # to satisfy any dependencies then use pip to overwrite it.
 
-install_package python-virtualenv
+install_package virtualenv-python
```
Comment (dtroyer) [inline tools/fixup_stuff.sh]: I have a feeling that these may not be needed any more for anyone, but for now you could just check ! is_clearlinux here and skip it.  Does that work?

File: lib/databases/mysql [hunk 9]
```diff
@@ -16,7 +16,8 @@ MYSQL_DRIVER=${MYSQL_DRIVER:-PyMySQL}
 register_database mysql
 
 # Linux distros, thank you for being incredibly consistent
-MYSQL=mysql
+#MYSQL=mysql
+MYSQL=mariadb
```
Comment (dtroyer) [inline lib/databases/mysql]: add a check to the line below to do this

File: lib/databases/mysql [hunk 10]
```diff
@@ -161,7 +164,9 @@ EOF
     elif is_ubuntu; then
         install_package mysql-server
     else
-        exit_distro_not_supported "mysql installation"
+	install_package mariadb-server
```
Comment (dtroyer) [inline lib/databases/mysql]: again, add the check using is_clearlinux to do this and leave the else clause in place as-is

File: lib/neutron [hunk 11]
```diff
@@ -359,7 +359,8 @@ function install_neutron_new {
 
     # L3 service requires radvd
     if is_service_enabled neutron-l3; then
```
Comment (dtroyer) [inline lib/neutron]: maybe just disable the l3 service?  Is the issue here that radvd isn't present in clear?

File: lib/databases/mysql (hunk 9 above)
Comment (ychen2u) [reply lib/databases/mysql]: Yes, it should be covered by is_clearlinux().

File: lib/neutron (hunk 11 above)
Comment (ychen2u) [reply lib/neutron]: radvd should have been included in the openstack-common.

**Comment by dtroyer** [comment]
These are being addressed individually

**Pull request closed without merging**